### ISSUE TITLE
Support of optim.lr_scheduler.ReduceLROnPlateau

### ISSUE THREAD

**Issue by vikmary** [williamFalcon/pytorch-lightning#298]
Hi,
I'm using pytorch-lightning as a base for a side project. My `configure_optimizers` looks this way:
```python
def configure_optimizers(self):
  """
  return whatever optimizers we want here
  :return: list of optimizers
  """
  optimizer = optim.Adam(self.parameters(), lr=self.params.learning_rate)
  scheduler = optim.lr_scheduler.ReduceLROnPlateau(optimizer,
                                                   mode='min',
                                                   factor=0.1,
                                                   patience=10,
                                                   min_lr=1e-6,
                                                   verbose=True)
  return [optimizer], [scheduler]
```

But that **doesn't work out of the box**, because the ReduceLROnPlateau scheduler needs ` scheduler.step(val_loss)` call instead of `scheduler.step(num_epochs)`.

The scheduler improves greatly nn convergence, it would nice to support it.

Thank you

**Comment by williamFalcon** [comment]
install from master and it’ll work

**Comment by vikmary** [comment]
Are you talking about this fix?
https://github.com/williamFalcon/pytorch-lightning/commit/e0c54067c1e694571c54086087104ae6b569ae1a

It doesn't solve the issue.

It should be `scheduler.step(val_loss, epochs=nb_epochs)`, not `scheduler.step(nb_epochs)`. Right now scheduler is changing learning rate based on epoch number, not validation loss.

**Comment by williamFalcon** [comment]
got it. thanks for pointing that out. want to submit a PR?
